src/status_im/ui2/screens/chat/composer/input.cljs

```diff
@@ -192,49 +192,59 @@
+                                                                          colors/white-opa-30)
+                             :placeholder (if cooldown-enabled?
+                                            (i18n/label :cooldown/text-input-disabled)
+                                            (i18n/label :t/type-a-message))
+                             :default-value (get @input-texts chat-id)
+                             :underline-color-android :transparent
+                             :auto-capitalize :sentences
+                             :auto-correct false
+                             :spell-check false
+                             :on-content-size-change on-content-size-change
+                             :on-selection-change (partial on-selection-change
+                                                           timeout-id
+                                                           last-text-change
+                                                           mentionable-users)
+                             :on-change
+                             (partial on-change
+                                      last-text-change
+                                      timeout-id
+                                      mentionable-users
+                                      refs
+                                      chat-id
+                                      sending-image)
+                             :on-text-input (partial on-text-input
+                                                     mentionable-users
+                                                     chat-id)}
         input-with-mentions (rf/sub [:chat/input-with-mentions])
-        children (fn []
-                   (if mentions-enabled?
-                     (map-indexed
-                      (fn [index [_ text]]
-                        ^{:key (str index "_" type "_" text)}
-                        [rn/text (when (= type :mention) {:style {:color colors/primary-50}})
-                         text])
-                      input-with-mentions)
-                     (get @input-texts chat-id)))]
+        children            (if mentions-enabled?
+                              (map-indexed (fn [index [text-type text]]
+                                             ;; This type is the clojure fn?
+                                             ;; Should it be in the destructuring vector?
+                                             ^{:key (str index "_" text-type "_" text)}
+                                             [rn/text
+                                              (when (= text-type :mention)
+                                                {:style {:color colors/primary-50}})
+                                              text])
+                                           input-with-mentions)
+                              (get @input-texts chat-id))]
     (reset! text-input-ref (:text-input-ref refs))
```

Comment:
Usually, debug atom updates in the body of the render function in a component is very hard, since they trigger re-renders. I prefer not to use updates this way, this component uses them very often, it seems hard to refactor, but I think it's worth it.

Also, usually we don't need that react refs are ratoms, using a regular atom usually is enough, so we can set the value and it doesn't trigger a re-render.

Reply:
100%, ideally refs should not be ratoms.